src/fileExplorerFile.ts

```diff
@@ -3,25 +3,52 @@ import { findCorrespondingXoppToPdf, openXournalppFile } from "./xoppActions";
 import { TFile } from "obsidian";
 
 export function addOpenInXournalpp(plugin: XoppPlugin) {
+    plugin.app.workspace.onLayoutReady(() => {
+        observeFileExplorer(plugin); // Start observing once the layout is ready
+        applyXournalppTags(plugin); // Apply the tags initially
+    });
+}
+
+function observeFileExplorer(plugin: XoppPlugin) {
     let fileExplorers = plugin.app.workspace.getLeavesOfType("file-explorer");
 
     fileExplorers.forEach(fileExplorer => {
-        let allFiles = (fileExplorer.view as any)?.fileItems;
+        let container = (fileExplorer.view as any)?.containerEl;
+
+        if (container) {
+            // Observe changes in the file explorer DOM
+            const observer = new MutationObserver(() => {
+                applyXournalppTags(plugin); // Reapply tags whenever there's a DOM change
+            });
+
+            // Observe changes to child elements (subtree and childList options)
+            observer.observe(container, { childList: true, subtree: true });
+        }
+    });
+}
+
+function applyXournalppTags(plugin: XoppPlugin) {
+    let fileExplorers = plugin.app.workspace.getLeavesOfType("file-explorer");
+
+    fileExplorers.forEach(fileExplorer => {
+        let allFiles: { [key: string]: { tagEl: HTMLElement } } = (fileExplorer.view as any)?.fileItems;
 
-        const files:Array<Array<HTMLElement|TFile>> = [];
         Object.entries(allFiles).forEach(([filePath, value]) => {
             if (filePath.endsWith(".pdf")) {
-                let xoppFile = findCorrespondingXoppToPdf(filePath as string, plugin)
-                if (xoppFile) files.push([(value as any).tagEl as HTMLElement, xoppFile]);
+                let xoppFile = findCorrespondingXoppToPdf(filePath, plugin);
+                if (xoppFile && value?.tagEl) {
+                    const tagEl = value.tagEl as HTMLElement;
+
+                    if (tagEl && !tagEl.classList.contains("clickable-tag")) {
+                        tagEl.innerText = "X++";
+                        tagEl.classList.add("clickable-tag");
+                        tagEl.onclick = () => {
+                            openXournalppFile(xoppFile, plugin);
+                        };
+                        console.log("TagEl successful for file:", filePath);
```

Comment:
Hey Dan

Just saw that there was still a `console.log()` in the code. Would you be able to remove that, I don't think that is a necessary log message for the user, right?

Reply:
Done.

I had an idea about a new feature - syncing renamed files (I had such situations a couple of times, when I needed to rename a file, but it was linked to a different xopp then). Do you have any ideas on that?